### PR TITLE
ContextMenu and ContextMenuItem Suppressor

### LINKED ISSUE #19: Suppress warnings for methods decorated with ContextMenu
See the image in:

https://twitter.com/JfranMora/status/1224742294483619842?s=20

The method `ToggleInstance` is greyed out as it's statically unused. We should also cover that case in our suppressor.

### PR DESCRIPTION
**Pull request by AnBucyk**
Fixes #19

#### Checklist

- [X] I have read the [Contribution Guide](/CONTRIBUTING.md) ;
- [X] There is an approved issue describing the change when contributing a new analyzer or suppressor ;
- [x] I have added tests that prove my fix is effective or that my feature works ; (N/A, need suppressor test framework)
- [X] I have added necessary documentation (if appropriate) ;

#### Short description of what this resolves:
This PR adds a new suppressor for methods and fields decorated with the ContextMenu or ContextMenuItem attributes, preventing them from being marked as unused or made read-only.

### CODE REVIEW

**Comment by jbevain** [comment]
Thanks! One thing that should be supressed as well, the second argument to ContextMenuItem is the name of the function that is invoked:

```
[ContextMenuItem("Reset Health", "ResetHealth")]
private int health;

private void ResetHealth()
{
    health = 100;
}
```

The function ResetHealth could trigger a IDE0055 that becomes incorrect.

**Comment by AnBucyk** [comment]
@jbevain good catch! I'll modify this to support that scenario.

File: src/Microsoft.Unity.Analyzers/ContextMenuSuppressor.cs [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+				if (!(member is IFieldSymbol fieldSymbol))
+					continue;
+
+				var attributes = fieldSymbol.GetAttributes().Where(a => a.AttributeClass.Matches(typeof(UnityEngine.ContextMenuItemAttribute)));
+
+				if (!attributes.Any())
+					continue;
+
+				foreach (var attribute in attributes)
+					if (symbol.Name == attribute.ConstructorArguments[1].Value.ToString())
```
Comment (AnBucyk) [inline src/Microsoft.Unity.Analyzers/ContextMenuSuppressor.cs]: @jbevain Is there a better way to do this that doesn't rely on a string comparison?

Comment (AnBucyk) [reply src/Microsoft.Unity.Analyzers/ContextMenuSuppressor.cs]: That's also a nasty nullref risk there, I think. Fixing now.

File: doc/USP0010.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+IDE0051 - Remove unused private members
+
+## Examples of code that produces a suppressed diagnostic
+```csharp
+using UnityEngine;
+
+class Camera : MonoBehaviour
+{
+	[ContextMenuItem("Foo", "Bar")]
```
Comment (sailro) [inline doc/USP0010.md]: Indent

File: doc/USP0011.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+IDE0044 - Make field readonly
+
+## Examples of code that produces a suppressed diagnostic
+```csharp
+using UnityEngine;
+
+class Camera : MonoBehaviour
+{
+	[ContextMenuItem("Foo", "Bar")]
```
Comment (sailro) [inline doc/USP0011.md]: Indent

**Comment by sailro** [comment]
Merged master, given we have now the test-infrastructure for testing suppressors

**Pull request merged**